qiskit/optimization/algorithms/slsqp_optimizer.py

```diff
@@ -218,9 +220,14 @@ def _minimize(x_0: np.array) -> Tuple[np.array, Any]:
         # actual optimization goes here
         result = self.multi_start_solve(_minimize, problem)
 
+        # check for feasibility
+        status = OptimizationResultStatus.SUCCESS if problem.is_feasible(result.x) \
+            else OptimizationResultStatus.INFEASIBLE
+
```

Comment:
These three lines bother me a lot. All the optimizers have them. Please consider a method `get_feasibility_status(x) -> OptimizationResultStatus` in the base class `OptimizationAlgorithm` that would do the same.
Then when a result is created just replace `status=self.get_feasibility_status(x)`, so there's no need to create a dedicated variable to keep status (I hope so).
Correct me if this is impractical.